### PR TITLE
OdroidC2 enabled USB_MASS_STORAGE

### PR DESCRIPTION
**Pull request by danielfmo**
as requested here: https://github.com/Raybuntu/LibreELEC.tv/issues/44

### CODE REVIEW

**Review by Ray-future — APPROVED** [review]
Thanks I will test if it builds and then merge it. This change should be upstreamed to LibreELEC too.

File: projects/Odroid_C2/linux/linux.aarch64.conf [hunk 1]
```diff
@@ -2712,7 +2712,7 @@ CONFIG_USB_GADGET_STORAGE_NUM_BUFFERS=2
 # CONFIG_USB_G_NCM is not set
 # CONFIG_USB_GADGETFS is not set
 # CONFIG_USB_FUNCTIONFS is not set
-# CONFIG_USB_MASS_STORAGE is not set
+USB_MASS_STORAGE=y
```
Comment (Ray-future) [inline projects/Odroid_C2/linux/linux.aarch64.conf]: It has to be CONFIG_USB_MASS_STORAGE not USB_MASS_STORAGE ;)

**Review by Ray-future — CHANGES_REQUESTED** [review]

**Comment by Ray-future** [comment]
@danielfmo I've tested this PR but unfortunately it requires some additional config changes. I'm not sure what additionally is required to solve issue #44 .
So can you please go through the config changes and then add those too to this PR?

**Comment by danielfmo** [comment]
@mkmt: as stated by Raybuntu, those configurations are not enough, can you please add what is missing? 
Or at lease share some sources.

**Comment by mkmt** [comment]
@danielfmo
USB_MASS_STORAGE must be set as a module (all configuration is done while loading module),

dependencies for this module are:
USB_SUPPORT (already done)
USB_GADGET (already done)
BLOCK (already done)

missing:
USB_CONFIGFS (think as a module as well)
USB_GADGETFS (think as a module as well)
USB_MASS_STORAGE (as a module)

It compiles on my laptop which doesn't have OTG port - so can't check if that works, but it compiles without problems.

There is website where is described how USB_MASS_STORAGE works on different platform, which should work on C2 as well: http://linux-sunxi.org/USB_Gadget/Mass_storage

**Comment by Ray-future** [comment]
This is what I have when using oldconfig: http://sprunge.us/LjCR

**Comment by Ray-future** [comment]
Odroid C2 was moved to the Amlogic project in LibreELEC. I suggest you send your PR to official LE.

**Pull request closed without merging**